### ISSUE TITLE
Crash with large double value

### ISSUE THREAD

**Issue by robertodtg** [paralect/robomongo#748]
The robomongo dies when you view a document that contains large Double field.

Test sequence (create one collection named "test")

// value 1.234 * 10^56
  db.test.insert({test: Number(1.234E56)})

// Robomongo crash here
  db.test.find()

**Comment by stennie** [comment]
@robertodtg I just tried your test with Robomongo 0.8.4 (OS X) and couldn't reproduce the crash. I tried toggling between tree/table/text view and that also didn't make a difference. Can you include more details on your O/S, version of Robomongo, and version of MongoDB?

More detailed steps to reproduce or a stack trace / error log from the crash would be helpful.

Thanks!

**Comment by gavruk** [comment]
I've reproduced this on Windows 8.1

**Comment by stennie** [comment]
Crashes for me on Windows 7 as well (can't repro on OS X).

I get a BEX exception: c0000409 (Stack buffer overflow)

Perhaps an issue with Windows release being 32-bit (OS X is 64-bit). Would be worth testing on Linux 32/64-bit for comparison.

**Comment by gavruk** [comment]
This issue reproduced on my 64-bit Windows 8.1

**Comment by stennie** [comment]
@gavruk Were you running a 64-bit Robomongo build on Windows? The Windows 0.8.4 release version from robomongo.org is 32-bit.

**Comment by gavruk** [comment]
@stennie, ah, yes. Windows is 64bit, but Robomongo is 32bit

**Comment by stennie** [comment]
This causes a buffer overflow on 64-bit Linux 0.8.5 build (oddly, OS X is not affected)
```
/lib/x86_64-linux-gnu/libc.so.6( 0x7338f)[0x7f1ea7feb38f]
/lib/x86_64-linux-gnu/libc.so.6(__fortify_fail 0x5c)[0x7f1ea8082c9c]
/lib/x86_64-linux-gnu/libc.so.6( 0x109b60)[0x7f1ea8081b60]
/lib/x86_64-linux-gnu/libc.so.6( 0x109069)[0x7f1ea8081069]
/lib/x86_64-linux-gnu/libc.so.6(_IO_default_xsputn 0xbc)[0x7f1ea7ff370c]
/lib/x86_64-linux-gnu/libc.so.6(__printf_fp 0x1849)[0x7f1ea7fc8b29]
/lib/x86_64-linux-gnu/libc.so.6(_IO_vfprintf 0x4aa3)[0x7f1ea7fc6793]
/lib/x86_64-linux-gnu/libc.so.6(__vsprintf_chk 0x84)[0x7f1ea80810f4]
/lib/x86_64-linux-gnu/libc.so.6(__sprintf_chk 0x7d)[0x7f1ea808104d]
./robomongo(Robomongo::BsonUtils::buildJsonString(mongo::BSONElement const&, std::string&, Robomongo::UUIDEncoding, Robomongo::SupportedTimes) 0x583)[0x52cac3]
./robomongo[0x545cff]
./robomongo(Robomongo::BsonTreeModel::BsonTreeModel(std::vector<boost::shared_ptr<Robomongo::MongoDocument>, std::allocator<boost::shared_ptr<Robomongo::MongoDocument> > > const&, QObject*) 0x161)[0x545f61]
./robomongo(Robomongo::OutputItemContentWidget::configureModel() 0x34)[0x509a34]
./robomongo(Robomongo::OutputItemContentWidget::setup(double) 0x13c)[0x50a4cc]
./robomongo(Robomongo::OutputItemContentWidget::OutputItemContentWidget(Robomongo::OutputWidget*, Robomongo::ViewMode, Robomongo::MongoShell*, QString const&, std::vector<boost::shared_ptr<Robomongo::MongoDocument>, std::allocator<boost::shared_ptr<Robomongo::MongoDocument> > > const&, Robomongo::MongoQueryInfo const&, double, QWidget*) 0x25b)[0x50aabb]
./robomongo(Robomongo::OutputWidget::present(Robomongo::MongoShell*, std::vector<Robomongo::MongoShellResult, std::allocator<Robomongo::MongoShellResult> > const&) 0x582)[0x508b82]
./robomongo(Robomongo::QueryWidget::displayData(std::vector<Robomongo::MongoShellResult, std::allocator<Robomongo::MongoShellResult> > const&, bool) 0x4c)[0x50513c]
./robomongo(Robomongo::QueryWidget::handle(Robomongo::ScriptExecutedEvent*) 0x176)[0x505be6]
/home/stennie/Qt5.3.2/5.3/gcc_64/lib/libQt5Core.so.5(QMetaMethod::invoke(QObject*, Qt::ConnectionType, QGenericReturnArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument) const 0x5c3)[0x7f1eaa5f48a3]
/home/stennie/Qt5.3.2/5.3/gcc_64/lib/libQt5Core.so.5(QMetaObject::invokeMethod(QObject*, char const*, Qt::ConnectionType, QGenericReturnArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument) 0x92b)[0x7f1eaa5fa1ab]
./robomongo(Robomongo::EventBusDispatcher::event(QEvent*) 0x161)[0x5260a1]
/home/stennie/Qt5.3.2/5.3/gcc_64/lib/libQt5Widgets.so.5(QApplicationPrivate::notify_helper(QObject*, QEvent*) 0xb4)[0x7f1ea9c1e724]
/home/stennie/Qt5.3.2/5.3/gcc_64/lib/libQt5Widgets.so.5(QApplication::notify(QObject*, QEvent*) 0x616)[0x7f1ea9c21d46]
/home/stennie/Qt5.3.2/5.3/gcc_64/lib/libQt5Core.so.5(QCoreApplication::notifyInternal(QObject*, QEvent*) 0x84)[0x7f1eaa5e9fc4]
./robomongo(Robomongo::EventBus::sendEvent(Robomongo::EventBusDispatcher*, Robomongo::EventWrapper*) 0x53)[0x522ce3]
./robomongo(Robomongo::EventBus::publish(Robomongo::Event*) 0x12e)[0x522e3e]
./robomongo(Robomongo::MongoShell::handle(Robomongo::ExecuteScriptResponse*) 0x2eb)[0x525cfb]
/home/stennie/Qt5.3.2/5.3/gcc_64/lib/libQt5Core.so.5(QMetaMethod::invoke(QObject*, Qt::ConnectionType, QGenericReturnArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument) const 0x5c3)[0x7f1eaa5f48a3]
/home/stennie/Qt5.3.2/5.3/gcc_64/lib/libQt5Core.so.5(QMetaObject::invokeMethod(QObject*, char const*, Qt::ConnectionType, QGenericReturnArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument, QGenericArgument) 0x92b)[0x7f1eaa5fa1ab]
./robomongo(Robomongo::EventBusDispatcher::event(QEvent*) 0x161)[0x5260a1]
/home/stennie/Qt5.3.2/5.3/gcc_64/lib/libQt5Widgets.so.5(QApplicationPrivate::notify_helper(QObject*, QEvent*) 0xb4)[0x7f1ea9c1e724]
/home/stennie/Qt5.3.2/5.3/gcc_64/lib/libQt5Widgets.so.5(QApplication::notify(QObject*, QEvent*) 0x616)[0x7f1ea9c21d46]
/home/stennie/Qt5.3.2/5.3/gcc_64/lib/libQt5Core.so.5(QCoreApplication::notifyInternal(QObject*, QEvent*) 0x84)[0x7f1eaa5e9fc4]
/home/stennie/Qt5.3.2/5.3/gcc_64/lib/libQt5Core.so.5(QCoreApplicationPrivate::sendPostedEvents(QObject*, int, QThreadData*) 0x258)[0x7f1eaa5ecb48]
/home/stennie/Qt5.3.2/5.3/gcc_64/lib/libQt5Core.so.5( 0x329253)[0x7f1eaa644253]
/lib/x86_64-linux-gnu/libglib-2.0.so.0(g_main_context_dispatch 0x254)[0x7f1ea6f02e04]
/lib/x86_64-linux-gnu/libglib-2.0.so.0( 0x49048)[0x7f1ea6f03048]
/lib/x86_64-linux-gnu/libglib-2.0.so.0(g_main_context_iteration 0x2c)[0x7f1ea6f030ec]
/home/stennie/Qt5.3.2/5.3/gcc_64/lib/libQt5Core.so.5(QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) 0x64)[0x7f1eaa644684]
/home/stennie/Qt5.3.2/5.3/gcc_64/lib/libQt5Core.so.5(QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) 0xcb)[0x7f1eaa5e81db]
/home/stennie/Qt5.3.2/5.3/gcc_64/lib/libQt5Core.so.5(QCoreApplication::exec() 0x85)[0x7f1eaa5ed0a5]
./robomongo(main 0xf0)[0x4f2100]
/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main 0xf5)[0x7f1ea7f99ec5]
./robomongo[0x4fb817]
```

**Comment by schetnikovich** [comment]
@robertodtg @gavruk This issue was fixed in just released [Robomongo 0.9.0 RC3](http://blog.robomongo.org/robomongo-0-9-0-rc3-released). Let us know how it works for you.

**Comment by gavruk** [comment]
I confirm that this issue is fixed in 0.9.0 RC3
